src/backend/types.ml

```diff
@@ -301,3 +330,198 @@ module Competition = struct
       ]
 end
 
+(* Artefact *)
+(* ************************************************************************* *)
+
+module YanCriterion = struct
+  type t = string * Ftw.Ranking.Algorithm.yan_weight [@@deriving yojson]
+
+  let ref, schema =
+    make_schema ()
+      ~name:"YanCriterion"
+      ~typ:(Obj Object)
+      ~additional_properties:(
+        obj @@ S.make_schema ()
+          ~properties:[
+            "yes", obj @@ S.make_schema ()
+              ~typ:int;
+            "alt", obj @@ S.make_schema ()
+              ~typ:int;
+            "no", obj @@ S.make_schema ()
+              ~typ:int;
+          ]
+      )
+end
+
+
+module ArtefactDescription = struct
+
+  module StrMap = Map.Make (String)
+
+  type t = {
+    artefact: string;
+    yan_criterion: YanCriterion.t list option;
+    algorithm_for_ranking: string option;
+  }[@@deriving yojson]
+
+  let ref, schema =
+    make_schema ()
+      ~name:"ArtefactDescription"
+      ~typ:(Obj Object)
+      ~description: {| artefact is either ranking or yan.
+        For a ranking artefact, ranking_algorithm property should be specified.
+        For a yan artefact, yan_criterion property should be set. |}
+      ~properties:[
+        "artefact", obj @@ S.make_schema ()
+          ~typ:string
+          ~enum:[`String "ranking"; `String "yan"];
+        "yan_criterion", obj @@ S.make_schema ()
+          ~typ:(Obj Array)
+          ~items:(ref YanCriterion.ref);
+        "algorithm_for_ranking", obj @@ S.make_schema ()
+          ~typ:string
+      ]
+      ~required:["artefact"]
```

Comment:
We really need to distinguish artefact descriptions from ranking algorithm in the API types: this is an important distinction that the lib makes, and in the future we want to be able to see what would happen to a competition's results if the ranking algorithm was changed: for that we need to be able to specify a ranking algorithm alone via the API (since it wouldn't make sense to try and change the artefact description for a competition that has already happened and for which we have the artefacts).